x/pool/ref_counted_pooled_string_array_test.go

```diff
@@ -32,24 +33,25 @@ func TestRefCountedPooledStringArray(t *testing.T) {
 	require.Equal(t, 4, len(arr.Get()))
 	require.Equal(t, int32(2), arr2.cnt.RefCount())
 	require.Equal(t, 4, len(arr2.Get()))
-	expected1 := []string{"foo", "bar", "baz", "cat"}
-	assertReturnedToStringArrayPool(t, pool, expected1, false)
+	assertReturnedToStringArrayPool(t, pool, arr.Get(), false)
```

Comment:
Hm I don't think you need to change the rest of the tests since you already set the reset fn to nil above. I generally prefer to assert against a very explicit value if possible so in this case I prefer to assert against an expected string array instead.